sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/PointWriter.scala

```diff
@@ -332,6 +337,62 @@ class PointWriter(container: CosmosAsyncContainer,
   }
   // scalastyle:on return
 
+  // scalastyle:on multiple.string.literals
+  private def patchBulkUpdateWithRetry(
+                                      partitionKeyValue: PartitionKey,
+                                      objectNode: ObjectNode): Unit = {
+
+      var exceptionOpt = Option.empty[Exception]
+      val patchBulkUpdateOperation = PatchBulkUpdateOperation(taskDiagnosticsContext,
+          CosmosItemIdentifier(objectNode.get(CosmosConstants.Properties.Id).asText(), partitionKeyValue))
+
+      for (attempt <- 1 to cosmosWriteConfig.maxRetryCount + 1) {
+          try {
+              patchBulkUpdateItem(container, partitionKeyValue, objectNode)
+              return
+          } catch {
+              case e: CosmosException if Exceptions.canBeTransientFailure(e.getStatusCode, e.getSubStatusCode) =>
+                  log.logWarning(
+                      s"patch update item $patchBulkUpdateOperation attempt #$attempt max remaining retries "
+                          + s"${cosmosWriteConfig.maxRetryCount + 1 - attempt}, encountered ${e.getMessage}")
+                  exceptionOpt = Option.apply(e)
+          }
+      }
+
+      log.logItemWriteFailure(patchBulkUpdateOperation, exceptionOpt.get)
+      assert(exceptionOpt.isDefined)
+      exceptionOpt.get.printStackTrace()
+      throw exceptionOpt.get
+  }
+  // scalastyle:on return
+
+  private[this] def patchBulkUpdateItem(container: CosmosAsyncContainer,
+                                    partitionKey: PartitionKey,
+                                    objectNode: ObjectNode): CosmosItemResponse[ObjectNode] = {
+      assert(cosmosPatchHelpOpt.isDefined)
+
+      val itemId = objectNode.get(CosmosConstants.Properties.Id).asText()
+      val patchBulkUpdateOperations = cosmosPatchHelpOpt.get.createCosmosPatchBulkUpdateOperations(objectNode)
+
+      container
+          .readItem(itemId, partitionKey, classOf[ObjectNode])
+          .flatMap(response => {
+              val updatedNode = cosmosPatchHelpOpt.get.patchBulkUpdateItem(Some(response.getItem), patchBulkUpdateOperations)
+              val cosmosItemRequestOptions = new CosmosItemRequestOptions().setIfMatchETag(response.getETag)
+              container.replaceItem(updatedNode, itemId, partitionKey, cosmosItemRequestOptions)
+          })
+          .asScala
+          .onErrorResume(throwable => {
+              throwable match {
+                  case e: CosmosException if Exceptions.isNotFoundException(e) =>
+                      val updatedNode = cosmosPatchHelpOpt.get.patchBulkUpdateItem(None, patchBulkUpdateOperations)
+                      container.createItem(updatedNode, partitionKey, new CosmosItemRequestOptions()).asScala
```

Comment:
This one can now result in 409 which would need to be retried again

Reply:
good catch, added checking for 409/412 as well